medmodels/medrecord/indexers.py

```
@@ -515,14 +602,26 @@ def __setitem__(  # noqa: C901
                 msg = "Invalid value type. Expected MedRecordValue"
                 raise ValueError(msg)
 
-            attributes = self._medrecord._medrecord.node(
-                self._medrecord.select_nodes(index_selection)
-            )
+            query_result = self._medrecord.query_nodes(index_selection)
 
-            for node in attributes:
-                for attribute in attributes[node]:
+            if isinstance(query_result, list):
+                attributes = self._medrecord._medrecord.node(query_result)
+
+                for node in attributes:
+                    for attribute in attributes[node]:
+                        self._medrecord._medrecord.update_node_attribute(
+                            [node], attribute, value
+                        )
+            elif query_result:
+                attributes = self._medrecord._medrecord.node([query_result])[
+                    query_result
+                ]
+
+                for attribute in attributes:
                     self._medrecord._medrecord.update_node_attribute(
-                        [node], attribute, value
+                        [query_result],
+                        attribute,
+                        value,
                     )
 
             return None
```

Should we not have a warning or Error of some sort in case there is no query_result? (what happens when else: is triggered?)

---

I also thought about this a lot.

I don't think this should throw an Error (except for when the typing contract would otherwise be broken, e.g. in `__getitem__`). I also don't think this should log a warning (though I'm not 100% sure in this case). I'm thinking that warnings should be used where something caused by the user is happening that is either implicit or that the user cannot really check/comprehend. In this case however, the user could simply execute the query and see that the query matches no nodes.
Also there is nothing "bad" happening here. Exactly the node that is described in the users query (so None) is deleted/updated.

I'm leaning towards not adding a warning

---

Okay, I am convinced. It just felt weird to have and else: clause hanging and did not know whether that was intended.
It is true that not doing anything to the MedRecord should not result in a warning.